src/domain/checks/LegalFactDownloadMetadataChecks.ts

```diff
@@ -0,0 +1,65 @@
+import { pipe } from 'fp-ts/function';
+import * as RA from 'fp-ts/ReadonlyArray';
+import * as R from 'fp-ts/Reader';
+import * as P from 'fp-ts/Predicate';
+import { isDownloadRecord } from '../DownloadRecord';
+import { Record } from '../Repository';
+import { isLegalFactDownloadMetadataRecord, LegalFactDownloadMetadataRecord } from '../LegalFactDownloadMetadataRecord';
+import { GetNotificationDetailRecord, isGetNotificationDetailRecord } from '../GetNotificationDetailRecord';
+import * as DownloadRecordChecks from './DownloadRecordChecks';
+
+const hasCalledDownloadEndpointForLegalFactC = pipe(
+  R.Do,
+  R.apS('downloadRecordList', RA.filterMap(isDownloadRecord)),
+  R.apS('legalFactDownloadMetadataRecordList', RA.filterMap(isLegalFactDownloadMetadataRecord)),
+  R.map(({ downloadRecordList, legalFactDownloadMetadataRecordList }) =>
+    pipe(
+      legalFactDownloadMetadataRecordList,
+      RA.every(DownloadRecordChecks.metadataRecordMatchesDownloadRecordC(downloadRecordList))
+    )
+  )
+);
+
+const matchesIunC =
+  ({ output }: GetNotificationDetailRecord) =>
+  (legalFactDownloadMetadataRecord: LegalFactDownloadMetadataRecord) =>
+    pipe(output.statusCode === 200 && output.returned.iun === legalFactDownloadMetadataRecord.input.iun);
+
+const matchTimelineValuesC =
+  ({ output }: GetNotificationDetailRecord) =>
+  ({ input }: LegalFactDownloadMetadataRecord) =>
+    pipe(
+      output.statusCode === 200 &&
+        output.returned.timeline.some(({ legalFactsIds }) =>
+          legalFactsIds?.some(({ category, key }) => category === input.legalFactType && key === input.legalFactId)
+        )
+    );
+
+const matchesLegalFactDownloadMetadataRecordC =
+  (legalFactDownloadMetadataRecord: LegalFactDownloadMetadataRecord) =>
+  (getNotificationDetailRecord: GetNotificationDetailRecord) =>
+    legalFactDownloadMetadataRecord.output.statusCode === 200 &&
+    matchesIunC(getNotificationDetailRecord)(legalFactDownloadMetadataRecord) &&
+    matchTimelineValuesC(getNotificationDetailRecord)(legalFactDownloadMetadataRecord);
```

Comment:
Pick the one you prefer 🙂 

```suggestion
const matchesLegalFactDownloadMetadataRecordC =
  ({ input, output }: LegalFactDownloadMetadataRecord) =>
  (notificationRecord: GetNotificationDetailRecord) =>
    output.statusCode === 200 &&
    notificationRecord.output.statusCode === 200 &&
    notificationRecord.output.returned.iun === input.iun &&
    pipe(
      notificationRecord.output.returned.timeline,
      RA.exists(({ legalFactsIds }) =>
        pipe(
          legalFactsIds || [],
          RA.exists(({ category, key }) => category === input.legalFactType && key === input.legalFactId)
        )
      )
    );
```

Reply:
Let's keep it clean!